src/main/java/tech/ydb/io/r2dbc/result/YdbRow.java

```diff
@@ -16,42 +16,46 @@
 
 package tech.ydb.io.r2dbc.result;
 
+import java.util.List;
+
 import io.r2dbc.spi.Row;
 import io.r2dbc.spi.RowMetadata;
-import tech.ydb.io.r2dbc.parameter.YdbParameterResolver;
-import tech.ydb.table.result.ResultSetReader;
-import tech.ydb.table.result.ValueReader;
+import tech.ydb.table.values.OptionalValue;
+import tech.ydb.table.values.Value;
 
 /**
  * @author Egor Kuleshov
  */
 public final class YdbRow implements Row {
-    private final ResultSetReader resultSetReader;
-    private final int rowIndex;
+    private final YdbRowMetadata ydbRowMetadata;
+    private final List<Value<?>> values;
 
-    public YdbRow(ResultSetReader resultSetReader, int rowIndex) {
-        this.resultSetReader = resultSetReader;
-        this.rowIndex = rowIndex;
+    public YdbRow(YdbRowMetadata ydbRowMetadata, List<Value<?>> values) {
+        this.ydbRowMetadata = ydbRowMetadata;
+        this.values = values;
     }
 
     @Override
     public RowMetadata getMetadata() {
-        return new YdbRowMetadata(resultSetReader);
+        return ydbRowMetadata;
     }
 
     @Override
     public <T> T get(int index, Class<T> type) {
-        resultSetReader.setRowIndex(rowIndex);
-        ValueReader valueReader = resultSetReader.getColumn(index);
-        if (valueReader.getValue().asOptional().isPresent()) {
-            return YdbParameterResolver.resolveResult(valueReader, type);
```

Comment:
This was a big mistake, since the result can change asynchronously, and the indexes here could also change asynchronously. To avoid this and not synchronize with the resultSetReader, I decided to store values in the row, ValueReader cannot be stored because after changing the reading row, they also change.

Это была большая ошибка, так как результат может изменяться асинхронно, и также асинхронно могли бы изменяться и индексы здесь. Чтобы избежать этого и не синхронизироваться по resultSetReader, я решил хранить в строке значения, ValueReader хранить нельзя так как после изменения строки чтения, они тоже изменяются.